test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/AddressRegistrationTests.cs

```diff
@@ -147,13 +148,14 @@ public class AddressRegistrationTests
             {
                 host.Start();
 
-                var debugLog = testLogger.Messages.Single(log => log.LogLevel == LogLevel.Debug);
-                Assert.True(debugLog.Message.Contains("default"));
+                Assert.Equal(5000, host.GetPort());
+                Assert.Single(testLogger.Messages.Where(log => log.LogLevel == LogLevel.Debug &&
```

Comment:
Use the overload that takes a collection and a predicate, then you don't need the `Where`.